src/shogun/lib/DynamicArray.h

```diff
@@ -619,13 +620,26 @@ template <class T> class CDynamicArray :public CSGObject
 			m_array.resize_array(m_array.get_num_elements(), true);
 		}
 
+		CSGObject* clone()
+		{
+			CDynamicArray * cloned = (CDynamicArray*) CSGObject::clone();
```

Comment:
after our discussion I think it would be best to have cloned objects have a different allocated sizes. But equals should still return true. Can be achieved by not registering the parameters and treating them in post loading?